nomenclature/config.py

```diff
@@ -1,12 +1,44 @@
 from pathlib import Path
-from typing import Dict, Optional
-from pydantic import BaseModel
+from typing import Optional
+from pydantic import BaseModel, root_validator
 
 import yaml
+from git import Repo
 
 
 class CodeListConfig(BaseModel):
-    repository: Optional[Path]
+    repository: Optional[str]
+    hash: Optional[str]
+    release: Optional[str]
+
+    @root_validator()
+    def check_hash_and_release(cls, v):
+        if v.get("hash") and v.get("release"):
+            raise ValueError("Either 'hash' or 'release' can be provided, not both.")
+        return v
+
+    @property
+    def revision(self):
+        return self.hash or self.release or "main"
+
+    @property
+    def repository_name(self):
+        return self.repository.split("/")[-2].split(".")[0]
+
+    def fetch_repo(self, to_path="./common-definitions"):
```

Comment:
Why is the *to_path* hard-coded?

Reply:
That was a left over from initial development, it's not removed.